xbstrap/vcs_utils.py

```diff
@@ -223,7 +228,12 @@ def fetch_repo(cfg, src, subdir, *, ignore_mirror=False, bare_repo=False):
             else:
                 subprocess.check_call([git, "init"] + b_args, cwd=source_dir)
             # We always set the remote to the true remote, not a mirror.
-            subprocess.check_call([git, "remote", "add", "origin", source["git"]], cwd=source_dir)
+            subprocess.check_call([git, "remote", "add", "origin", git_sources[0]], cwd=source_dir)
+
+            for num, fallback in enumerate(git_sources[1:]):
```

Comment:
i'm not sure I like splitting this array like this - it hard-codes the assumption of one element in the mirror list.

do we even need to handle fallbacks specially?  why not just insert all sources as remotes and save the list of names for further use

Reply:
We are guaranteed to have at least one element in the mirror list, as that's enforced by the schema. Having the primary remote be named `origin` is a convenience choice for the user that might use git inside the cloned package.

Reply:
hm, doesn't the schema accept `[]`?

Reply:
Fair point. I amended the schema to require at least one list entry and added an error message if `[]` is passed.